aten/src/ATen/native/transformers/cuda/attention_backward.cu

```diff
@@ -193,28 +192,70 @@ std::tuple<at::Tensor, at::Tensor, at::Tensor> _efficient_attention_backward(
   int64_t N = key.size(1);
   int64_t nH = query.size(2);
   int64_t K = query.size(3);
+  int64_t Kv = value.size(3);
+
+  const bool bias_requires_grad = bias.has_value() && bias->requires_grad();
 
-  // It does not make sense to use that in practice,
-  // but let's still make sure we are correct
-  // As we iterate through keys first, we skip
-  // keys with no query associated, so they are not
-  // initialized
-  bool grad_kv_needs_init = causal && N > M;
-  at::Tensor grad_q, grad_k, grad_v;
-  int8_t gQKV_strideM_multiplier = 1;
+  at::Tensor grad_q, grad_k, grad_v, grad_bias;
   grad_q = at::empty(query.sizes(), query.options());
-  grad_k = grad_kv_needs_init ? at::zeros(key.sizes(), key.options())
-                              : at::empty(key.sizes(), key.options());
-  grad_v = grad_kv_needs_init ? at::zeros(value.sizes(), value.options())
-                              : at::empty(value.sizes(), value.options());
+  grad_k = at::empty(key.sizes(), key.options());
```

Comment:
Removed the chunk optimization: see discussion here: https://github.com/pytorch/pytorch/pull/96880